fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneIndexQueryPlan.java

```diff
@@ -68,82 +62,28 @@ public class LuceneIndexQueryPlan extends RecordQueryIndexPlan implements PlanWi
-                                @Nullable PlanOrderingKey planOrderingKey, @Nullable List<KeyExpression> storedFields) {
+    protected LuceneIndexQueryPlan(@Nonnull String indexName, @Nonnull LuceneScanParameters scanParameters,
+                                   @Nonnull FetchIndexRecords fetchIndexRecords, boolean reverse,
+                                   @Nullable PlanOrderingKey planOrderingKey, @Nullable List<KeyExpression> storedFields) {
         super(indexName, null, scanParameters, IndexFetchMethod.SCAN_AND_FETCH, fetchIndexRecords, reverse, false);
         this.planOrderingKey = planOrderingKey;
         this.storedFields = storedFields;
     }
 
-    /**
-     * Override here to have specific logic to build the {@link QueryResult} for lucene auto-complete and spell-check results.
-     * For these 2 scan types, results are returned as partial records that have only the grouping keys, primary key,
-     * and the text fields indexed by Lucene which have matches with the search key.
-     * So other fields than the matching text field won't be able to be read from the partial records.
-     * For an indexed record that have matches from multiple text fields, matching suggestions are returned as multiple partial records with different text fields populated,
-     * which share the same grouping keys and primary key, so they are considered as multiple results.
-     */
     @Nonnull
     @Override
-    @SuppressWarnings("PMD.CloseResource")
-    public <M extends Message> RecordCursor<QueryResult> executePlan(@Nonnull FDBRecordStoreBase<M> store,
-                                                                     @Nonnull EvaluationContext context,
-                                                                     @Nullable byte[] continuation,
-                                                                     @Nonnull ExecuteProperties executeProperties) {
+    public <M extends Message> RecordCursor<FDBQueriedRecord<M>> fetchIndexRecords(@Nonnull final FDBRecordStoreBase<M> store,
+                                                                                   @Nonnull final EvaluationContext evaluationContext,
+                                                                                   @Nonnull final Function<byte[], RecordCursor<IndexEntry>> entryCursorFunction,
+                                                                                   @Nullable final byte[] continuation,
+                                                                                   @Nonnull final ExecuteProperties executeProperties) {
         final RecordMetaData metaData = store.getRecordMetaData();
         final Index index = metaData.getIndex(indexName);
         final Collection<RecordType> recordTypes = metaData.recordTypesForIndex(index);
         if (recordTypes.size() != 1) {
             throw new RecordCoreException("No lucene index should span multiple record types");
```

Comment:
If this error check is all that's left of the override, can it be moved from runtime into the planner and the override done away with altogether? Or pulled out into a method used for effect here and for value in the spell-checker's override?

Reply:
good point -- will see if I can improve that

Reply:
done